CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc

```diff
@@ -11,64 +11,43 @@ class X2Helper_Infiltration extends Object config(Infiltration);
 
 var config int PERSONNEL_INFIL;
 var config int PERSONNEL_DETER;
-var config float OVERLOADED_MULT_1;
-var config float OVERLOADED_MULT_2;
+var config array<float> OVERLOADED_MULT;
 
 var config array<int> RANKS_DETER;
 
-static function int GetSquadInfiltration(array<StateObjectReference> Soldiers)
+static function int GetSquadInfiltration(array<StateObjectReference> Soldiers, XComGameState_CovertAction CovertAction)
 {
 	local StateObjectReference	UnitRef;
-	local int	TotalInfiltration, FinalInfiltration, SquadSize, i;
-	
-	`LOG(" --- ");
+	local int	TotalInfiltration, FinalInfiltration, SquadSize, MaxSize;
+
+	MaxSize = GetRequiredStaffSlots(CovertAction);
+	MaxSize = MaxSize + (`XCOMHQ.HasSoldierUnlockTemplate('InfiltrationSize1') ? 1 : 0) + (`XCOMHQ.HasSoldierUnlockTemplate('InfiltrationSize2') ? 1 : 0);
```

Comment:
Save `XCOMHQ` to a variable. Also I don't like `MaxSize` name - maybe `NumUpgradedSlots`? Also, `+=`

Reply:
Why would you save XCOMHQ to a variable here? Why would you ever do it, actually? Using the macro seems to be much easier in all circumstances.